scalafix-rules/src/main/scala/scalafix/internal/rule/RemoveUnused.scala

```diff
@@ -153,6 +153,15 @@ class RemoveUnused(config: RemoveUnusedConfig)
         case Term.Function(List(param @ Term.Param(_, name, _, _)), _)
             if isUnusedParam(name.pos) =>
           Patch.replaceTree(param, "_")
+        case Term.Function(List(param @ Term.Param(_, name, _, _)), _)
+            if isUnusedParam(posExclParens(param)) =>
+          Patch.replaceTree(name, "_")
+        case Term.Function(params, _) =>
+          params.collect {
+            case param @ Term.Param(_, name, _, _)
+                if isUnusedParam(param.pos) =>
+              Patch.replaceTree(name, "_")
+          }.asPatch
```

Comment:
As you leave comments and we discussed in this PR, I'm a bit worrying that the reasons why each condition is chosen are not very clear (from the code, at least):

-  isUnusedParam(name.pos) or isUnusedParam(param.pos) or isUnusedParam(posExclParen(param))
- replaceTree(name, "_") or replaceTree(param, "_")
- match against 1-arity or n-arity function

Rather, it might be better to consolidate, something like:

```suggestion
        case Term.Function(params, _) =>
          params.collect {
            case param @ Term.Param(_, name, declType, _)
                if isUnusedParam(name.pos) || isUnusedParam(
                  posExclParens(param)
                ) =>
              Patch.replaceTree(if (declType.isDefined) name else param, "_")
          }.asPatch
```

What do you think?

Reply:
This will traverse Function trees a bit more.
Or, adding this kind of if-guard might be more user-friendly?

```scala
case Term.Function(params, _) if config.params =>
```

Reply:
I agree the code is complex, but I think refactoring everything into one `case` with a nested branch does not make the 3 different use cases and their gotchas explicit. I added 1968ecfc096e07ecf9264e83ccefd70d9d7ab28b to document why the last `case` is not enough.